pkg/resources/resources_manager_test.go

```diff
@@ -0,0 +1,295 @@
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(rm.configList)).To(Equal(2))
+		})
+		It("non existing config file", func() {
+
+			rm := &resourceManager{configFile: "/tmp/config.json"}
+			err := rm.ReadConfig()
+			Expect(err).To(HaveOccurred())
+		})
+		It("Read invalid config file", func() {
+			configData := `{"configList": [{
+             "resourceName": "hca_shared_devices_a",
+             "rdmaHcaMax": 1000,
+             "devices": ["ib0", "ib1"]
+           }}},
+        ]}`
+			fs := &utils.FakeFilesystem{
+				Dirs: []string{"tmp"},
+				Files: map[string][]byte{
+					"tmp/config.json": []byte(configData),
+				},
+			}
+			defer fs.Use()()
+
+			rm := &resourceManager{configFile: fs.RootDir + "/tmp/config.json"}
+			err := rm.ReadConfig()
+			Expect(err).To(HaveOccurred())
+		})
+	})
+	Context("ValidateConfigs", func() {
+		It("Valid config list", func() {
+			var configlist []*types.UserConfig
+			rm := &resourceManager{}
+
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   100,
+				Devices:      []string{"ib0"}})
+
+			rm.configList = configlist
+			err := rm.ValidateConfigs()
+			Expect(err).ToNot(HaveOccurred())
+		})
+		It("Validate empty config list", func() {
+			rm := &resourceManager{}
+			err := rm.ValidateConfigs()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("no resources configuration found"))
+		})
+		It("resources with invalid name", func() {
+			var configlist []*types.UserConfig
+			rm := &resourceManager{}
+
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config$$",
+				RdmaHcaMax:   100,
+				Devices:      []string{"ib0"}})
+
+			rm.configList = configlist
+			err := rm.ValidateConfigs()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("error: resource name \"test_config$$\" contains invalid characters"))
+		})
+		It("resources with repeated names", func() {
+			var configlist []*types.UserConfig
+			rm := &resourceManager{}
+
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   100,
+				Devices:      []string{"ib0"}})
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   500,
+				Devices:      []string{"ib1"}})
+
+			rm.configList = configlist
+			err := rm.ValidateConfigs()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("error: resource name \"test_config\" already exists"))
+		})
+		It("resources with invalid number of resources", func() {
+			var configlist []*types.UserConfig
+			rm := &resourceManager{}
+
+			configlist = append(configlist, &types.UserConfig{
+				ResourceName: "test_config",
+				RdmaHcaMax:   -100,
+				Devices:      []string{"ib0"}})
+
+			rm.configList = configlist
+			err := rm.ValidateConfigs()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("error: Invalid value for rdmaHcaMax < 0: -100"))
```

Comment:
Zero should be invalid number, can you add test case for it

Reply:
add case for 0 as valid case